diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -138,12 +141,31 @@ fun ASTNode.isFollowedByNewline() =
             it.treeNext.elementType == WHITE_SPACE && it.treeNext.text.contains("\n")
         } ?: false
 
+/**
+ * This function is similar to isFollowedByNewline(), but there may be a comment after the node
+ */
+fun ASTNode.isFollowedByNewlineWithComment() =
+        parent({ it.treeNext != null && it.treeNext.treeNext != null }, strict = false)?.let {
+            it.treeNext.treeNext.elementType == EOL_COMMENT && it.treeNext.treeNext.isFollowedByNewline()
+        } ?: false ||
+                parent({ it.treeNext != null }, strict = false)?.let {
+                    it.treeNext.run {
+                        when (elementType) {
+                            WHITE_SPACE -> text.contains("\n")
+                            EOL_COMMENT -> isFollowedByNewline()
+                            else -> false
+                        }
+                    }
+                } ?: false
+
 /**
  * Checks if there is a newline before this element. See [isFollowedByNewline] for motivation on parents check.
+ * Also there is
  */
 fun ASTNode.isBeginByNewline() =
         parent({ it.treePrev != null }, strict = false)?.let {
-            it.treePrev.elementType == WHITE_SPACE && it.treePrev.text.contains("\n")
+            it.treePrev.elementType == WHITE_SPACE && it.treePrev.text.contains("\n") ||
+                    (it.treePrev.elementType == IMPORT_LIST && it.treePrev.isLeaf() && it.treePrev.treePrev.isLeaf())
```

Comment:
Why do you need to check imports separately? Does it apply to other nodes in the beginning of the file too?

Reply:
I added KDOC for this